File: 02-programacion-basica/13-modularizacion/java/soluciones/RetoBoletaModular.java
public class RetoBoletaModular {
    public static void main(String[] args) {
        double subtotal = subtotal(25.0, 4);
        double descuento = descuento(subtotal);
        mostrar(subtotal, descuento, subtotal - descuento);
    }

    static double subtotal(double precio, int cantidad) { return precio * cantidad; }

    static double descuento(double subtotal) {
        return subtotal >= 100 ? subtotal * 0.10 : 0;
    }

    // La presentación se separa del cálculo para reducir responsabilidades mezcladas.
    static void mostrar(double subtotal, double descuento, double total) {
        System.out.printf("Subtotal: %.2f | Descuento: %.2f | Total: %.2f%n", subtotal, descuento, total);
    }
}

File: 02-programacion-basica/13-modularizacion/scala/soluciones/RetoBoletaModular.scala
def subtotal(precio: Double, cantidad: Int): Double = precio * cantidad
def descuento(subtotal: Double): Double = if subtotal >= 100 then subtotal * 0.10 else 0.0

def mostrar(subtotal: Double, descuento: Double, total: Double): Unit =
  // Presentar resultados queda separado de las reglas de cálculo.
  println(f"Subtotal: $subtotal%.2f | Descuento: $descuento%.2f | Total: $total%.2f")

@main def retoBoletaModular(): Unit =
  val sub = subtotal(25.0, 4)
  val desc = descuento(sub)
  mostrar(sub, desc, sub - desc)
